Add a method to easily delete settings_changes for a resource

bdunne opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1476365

Given:
```ruby
region.settings_for_resource
=> {:some_test_setting => {:setting => {:deeper => 1}, :other => 2}}
```

Calling:
```ruby
region.remove_settings_path_for_resource(:some_test_setting, :setting)
```

Results in:
```ruby
region.settings_for_resource
=> {:some_test_setting => {:other => 2}}
```

Fryguy commented:
What is the expected usage?  I basically ok with this, except I think the actual deletion method (lines 23-25) belongs on Vmdb::Settings itself.  Then, you can DRY up the .reload + broadcast call into a private method since we do the same thing on add_settings_for_resource

Fryguy commented:
Prefer

```ruby
settings_changes.where("key LIKE ?", "#{settings_path}%")
```

bdunne commented:
@Fryguy I added the method for reloading settings to reduce the duplication.  I'm not sure how I would move this to `Vmdb::Settings`.  I also added some tests.

Fryguy commented:
> I'm not sure how I would move this to Vmdb::Settings

I was thinking just move those 3 lines.  In particular, this method also technically belongs on Region and Zone, so perhaps it belongs in the configuration mixin?  Not sure why `add_settings_for_resource` wasn't there in the first place.

also what is the expected usage of this method

Fryguy commented on lib/vmdb/settings.rb:
```diff
@@ -55,6 +55,12 @@ def self.save!(resource, hash)
       apply_settings_changes(resource, deltas)
     end
 
+    def self.destroy!(resource, keys)
```
I don't like the name, but we can change it later.
